.github/workflows/rust.yml

```diff
@@ -82,6 +82,37 @@ jobs:
     - name: Run Clippy Lints
       run: cargo xtask clippy
 
+  check-features:
+    runs-on: ubuntu-22.04
+    env:
+      CARGO_INCREMENTAL: 0
+    steps:
+    # This repo is unstable and unnecessary: https://github.com/microsoft/linux-package-repositories/issues/34
+    - name: Disable packages.microsoft.com repo
+      run: sudo rm -f /etc/apt/sources.list.d/microsoft-prod.list
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ github.event.pull_request.head.sha }} # see omicron#4461
+    - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2.7.3
+      if: ${{ github.ref != 'refs/heads/main' }}
+    - name: Report cargo version
+      run: cargo --version
+    - name: Update PATH
+      run: source "./env.sh"; echo "PATH=$PATH" >> "$GITHUB_ENV"
+    - name: Print PATH
+      run: echo $PATH
+    - name: Print GITHUB_ENV
+      run: cat "$GITHUB_ENV"
+    - name: Install Pre-Requisites
+      run: ./tools/install_builder_prerequisites.sh -y
+    - name: Run Cargo Check (No Default Features)
+      run: cargo check --workspace --bins --tests --no-default-features
+    - name: Install cargo-hack
+      uses: taiki-e/install-action@cargo-hack
+    - name: Run Cargo Hack Check (Feature-Powerset, No-Dev-Deps)
+      timeout-minutes: 120 # 2 hours
+      run: cargo hack check --workspace --feature-powerset --no-dev-deps --exclude-features image-trampoline,image-standard
```

Comment:
Thanks for doing this! however this is going to be very hard for people to run locally. Could you make this an xtask command instead?

There are two options for cargo-hack -- a pre-built version-pinned binary, or as a library. Sadly it looks like cargo-hack isn't available as a library (there isn't a lib.rs in https://github.com/taiki-e/cargo-hack/tree/main/src) so the options are:

1. Pre-built binaries. Mac and Linux are available at https://github.com/taiki-e/cargo-hack/releases/tag/v0.6.28. For illumos we can either send the project a small PR, or build a binary similar to dpd-client etc. In either case, `cargo xtask` can download the binary and put it in the out dir we already have.
2. Working with Taiki to make cargo-hack a library as well, e.g. `lib-cargo-hack`.
3. Adapting bits of cargo-hack into cargo-xtask if it's reasonably small.

Reply:
Makes sense. Option 1 is probably best? But, I'll gander.

Reply:
Another option is to pin a version in CI, but just use whatever cargo-hack is in the PATH locally. (Similar to nextest.) I'm not quite sure if cargo-hack guarantees CLI stability the way nextest does.

Reply:
@sunshowers, to your last point, we should be able to tie into a `cargo-hack` release right? I've seen gh-actions install a specific version of `cargo-hack` or via semver. For local, I specify the semver version as part of the xtask.

Reply:
I'll push a variant of this and see what you think (as keeping a pre-build binary seems like too much here)?

Reply:
I will also make a PR to taiki's repo for cross-compiling, which we can then just download directly in the buildomat CI.